utils/utils.py

```diff
@@ -175,6 +175,7 @@ def get_key_def(key, config, default=None, expected_type=None, to_path: bool = F
             val = Path(to_absolute_path(val))
         except TypeError:
             logging.error(f"Couldn't convert value {val} to a pathlib.Path object")
+        expected_type = Path if expected_type == str else expected_type  # allows "str" and "Path" as expected_type
```

Comment:
change your `expected_type == str` for ` isinstance(expected_type, str)` it's possible that python dont like when you use `==` to compare type

Reply:
Makes sense, but doesn't seem to be working for some reason:
![image](https://user-images.githubusercontent.com/34774759/220743034-bd432d70-590a-4ca0-b016-9501e0420492.png)

Reply:
Actually, I understood: isinstance() check the type of str. It's type is a Python "type" object, so the "expected_type" is itself a type class of type "str", not a type class. That's super hard to explain and would need a bit more details, but not too important at this point. :)